integrators/symplectic_runge_kutta_nyström_integrator_test.cpp

```diff
@@ -69,51 +167,306 @@ TEST_F(SymplecticRungeKuttaNyströmIntegratorTest,
                 _1, _2, _3, &evaluations);
   IntegrationProblem<ODE> problem;
   problem.equation = harmonic_oscillator;
-  ODE::SystemState const initial_state = {{x_initial}, {v_initial}, t_initial};
+  ODE::SystemState const initial_state = {{q_initial}, {v_initial}, t_initial};
   problem.initial_state = &initial_state;
   problem.t_final = t_final;
   problem.append_state = [&solution](ODE::SystemState const& state) {
     solution.push_back(state);
   };
 
-  aba_integrator.Solve(problem, period / 10);
-  EXPECT_THAT(AbsoluteError(x_initial, solution.back().positions[0].value),
-              AllOf(Ge(3E-15 * Metre), Le(4E-15 * Metre)));
-  EXPECT_THAT(AbsoluteError(v_initial, solution.back().velocities[0].value),
-              AllOf(Ge(7E-8 * Metre / Second), Le(8E-8 * Metre / Second)));
-  EXPECT_EQ(t_final, solution.back().time.value);
+  integrator.Solve(problem, step);
+
   EXPECT_EQ(steps, solution.size());
-  EXPECT_EQ(14 * steps, evaluations);
+  switch (integrator.composition) {
+    case kBA:
+    case kABA:
+      EXPECT_EQ(steps * integrator.evaluations, evaluations);
+      break;
+    case kBAB:
+      EXPECT_EQ(steps * integrator.evaluations + 1, evaluations);
+      break;
+    default:
+      LOG(FATAL) << "Invalid composition";
+  }
+  Length q_error;
+  Speed v_error;
+  for (int i = 1; i <= steps; ++i) {
```

Comment:
It's a bit odd to go from 1 to steps here and compute i - 1 below.  I would go from 0 to steps - 1.